Index: settings.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/settings.gradle b/settings.gradle.kts
rename from settings.gradle
rename to settings.gradle.kts
--- a/settings.gradle	(revision 98cf54f37fe8ea2a48fc3d147c14c6186852a76f)
+++ b/settings.gradle.kts	(date 1695111339753)
@@ -3,7 +3,7 @@
         google()
         mavenCentral()
         gradlePluginPortal()
-        maven { url "https://jitpack.io" }
+        maven { setUrl("https://jitpack.io")  }
     }
 }
 dependencyResolutionManagement {
@@ -12,28 +12,28 @@
         google()
         mavenCentral()
         gradlePluginPortal()
-        maven { url "https://jitpack.io" }
+        maven { setUrl("https://jitpack.io")  }
     }
 }
 
-println "Start All"
+println("Start All")
 
-println "rootProject=" + rootProject
-println "rootDir    =" + rootDir
+println("rootProject=$rootProject")
+println("rootDir    =$rootDir")
 
-println "settings.dir=" + settingsDir
-println "settings.dir.parent=" + settingsDir.parent
+println("settings.dir=$settingsDir")
+println("settings.dir.parent=" + settingsDir.parent)
 
-include ':thirdlib'
-rootProject.name = 'MiniApp'
-include ':app'
+rootProject.name = "MiniApp"
+include (":thirdlib")
+include (":app")
 
 //setBinding(new Binding([gradle: this]))
-//evaluate(new File(settingsDir, 'sub_flutter/.android/include_flutter.groovy'))
-//include ':sub_flutter'
+//evaluate(new File(settingsDir, "sub_flutter/.android/include_flutter.groovy"))
+//include ":sub_flutter"
 
 
-include ':compose'
-include ':other'
-include ':common'
-include ':lib_jar'
+include (":compose")
+include (":other")
+include (":common")
+include (":lib_jar")
Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/build.gradle b/build.gradle.kts
rename from build.gradle
rename to build.gradle.kts
--- a/build.gradle	(revision 98cf54f37fe8ea2a48fc3d147c14c6186852a76f)
+++ b/build.gradle.kts	(date 1695111922585)
@@ -1,17 +1,19 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
 plugins {
-    id 'com.android.application' version '8.1.1' apply false
-    id 'com.android.library' version '8.1.1' apply false
-    id 'org.jetbrains.kotlin.android' version '1.8.10' apply false
-    id 'com.google.dagger.hilt.android' version '2.48' apply false
-    id 'com.google.devtools.ksp' version '1.8.10-1.0.9' apply false
+    id("com.android.application") version "8.1.1" apply false
+    id("com.android.library") version "8.1.1" apply false
+    id("org.jetbrains.kotlin.android") version "1.8.10" apply false
+    id("com.google.dagger.hilt.android") version "2.48" apply false
+    id("com.google.devtools.ksp") version "1.8.10-1.0.9" apply false
 }
-apply from: file('lib.gradle')
-apply from: file(rootProject.rootDir.absolutePath + "/custom-gradle/check-style.gradle")
+apply {
+    from(file("lib.gradle"))
+    from(file(rootProject.rootDir.absolutePath + "/custom-gradle/check-style.gradle"))
+}
 
-tasks.register('clean', Delete) {
-    delete rootProject.buildDir
-    delete rootProject.rootDir.absolutePath + "/final.apk"
-    delete rootDir.absolutePath + "/local_repo/"
+tasks.register("clean", Delete::class.java) {
+    delete(rootProject.buildDir)
+    delete(rootProject.rootDir.absolutePath + "/final.apk")
+    delete(rootDir.absolutePath + "/local_repo/")
 }
\ No newline at end of file
Index: app/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/build.gradle b/app/build.gradle.kts
rename from app/build.gradle
rename to app/build.gradle.kts
--- a/app/build.gradle	(revision 98cf54f37fe8ea2a48fc3d147c14c6186852a76f)
+++ b/app/build.gradle.kts	(date 1695112253068)
@@ -1,29 +1,29 @@
 plugins {
-    id 'com.android.application'
-    id 'org.jetbrains.kotlin.android'
-    id 'com.google.devtools.ksp'
-    id 'project-report'
-    id 'dagger.hilt.android.plugin'
+    id("com.android.application")
+    id("org.jetbrains.kotlin.android")
+    id ("com.google.devtools.ksp")
+    id ("project-report")
+    id("dagger.hilt.android.plugin")
 }
 import java.text.SimpleDateFormat
 
-def ext = rootProject.ext
-def buildTime = new SimpleDateFormat("yyMMddHHmm").format(new Date())
+val ext = rootProject.ext
+val buildTime = new SimpleDateFormat("yyMMddHHmm").format(new Date())
 android {
-    compileSdk ext.compileSdkVersion
+    compileSdk = ext.compileSdkVersion
 
     defaultConfig {
-        applicationId "com.engineer.android.mini"
-        minSdk ext.minSdkVersion
-        targetSdk ext.targetSdkVersion
-        versionCode 1
-        versionName "1.0"
+        applicationId = "com.engineer.android.mini"
+        minSdk = ext.minSdkVersion
+        targetSdk = ext.targetSdkVersion
+        versionCode = 1
+        versionName = "1.0"
 
-        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
-        resourceConfigurations += ['zh-rCN', 'xxhdpi']
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+        resourceConfigurations += ["zh-rCN", "xxhdpi"]
 
         ndk {
-            abiFilters 'arm64-v8a', 'armeabi-v7a'
+            abiFilters "arm64-v8a", "armeabi-v7a"
         }
 
         buildConfigField "Boolean", "enable_log", "false"
@@ -34,7 +34,7 @@
                                 extract_native_libs: "true",
                                 activity_exported  : "true"]
         vectorDrawables {
-            useSupportLibrary true
+            useSupportLibrary = true
         }
 
 
@@ -52,33 +52,33 @@
     buildTypes {
         release {
             //混淆
-            minifyEnabled true
+            minifyEnabled = true
             //所以尽可能的减少第三方的使用 也是可以降低混淆的难度
             //Zipalign优化
-            zipAlignEnabled true
+            zipAlignEnabled = true
             // 移除无用的resource文件
-            shrinkResources true
-            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
-            signingConfig signingConfigs.release
+            shrinkResources = true
+            proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro"_
+            signingConfig(signingConfigs.release_
         }
         debug {
-            debuggable true
-            minifyEnabled false
-            signingConfig signingConfigs.release
+            debuggable =true
+            minifyEnabled= false
+            signingConfig(signingConfigs.release)
         }
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_17
-        targetCompatibility JavaVersion.VERSION_17
+        sourceCompatibility = JavaVersion.VERSION_17
+        targetCompatibility = JavaVersion.VERSION_17
     }
     kotlinOptions {
         jvmTarget = "17"
     }
     buildFeatures {
-        dataBinding true
-        viewBinding true
-        aidl true
-        compose true
+        dataBinding = true
+        viewBinding =true
+        aidl = true
+        compose = true
     }
     composeOptions {
         kotlinCompilerExtensionVersion = "1.4.3"
@@ -86,10 +86,10 @@
 
     sourceSets {
         main {
-//            aidl.srcDirs = ['src/main/aidl']
+//            aidl.srcDirs = ["src/main/aidl"]
         }
         local {
-            manifest.srcFile 'src/local/AndroidManifest.xml'
+            manifest.srcFile "src/local/AndroidManifest.xml"
         }
     }
     flavorDimensions "channel", "type"
@@ -133,13 +133,13 @@
 
 
     }
-    namespace 'com.engineer.android.mini'
+    namespace "com.engineer.android.mini"
     lint {
         checkReleaseBuilds false
     }
     packagingOptions {
         resources {
-            excludes += '/META-INF/{AL2.0,LGPL2.1}'
+            excludes += "/META-INF/{AL2.0,LGPL2.1}"
         }
         jniLibs {
             useLegacyPackaging false
@@ -156,52 +156,52 @@
 }
 
 dependencies {
-    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation fileTree(dir: "libs", include: ["*.jar"])
 
 
     implementation "androidx.appcompat:appcompat:$appcompat"
     implementation "androidx.core:core-ktx:$core_ktx"
     implementation "androidx.constraintlayout:constraintlayout:$constraint_layout"
-    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
+    implementation "androidx.legacy:legacy-support-v4:1.0.0"
     implementation "com.google.android.material:material:$material"
     implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
     implementation "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
 
-    implementation 'com.github.bumptech.glide:glide:4.15.1'
-    implementation 'androidx.preference:preference-ktx:1.2.1'
+    implementation "com.github.bumptech.glide:glide:4.15.1"
+    implementation "androidx.preference:preference-ktx:1.2.1"
 
-    implementation 'com.facebook.stetho:stetho:1.6.0'
-    implementation 'com.facebook.stetho:stetho-okhttp3:1.6.0'
+    implementation "com.facebook.stetho:stetho:1.6.0"
+    implementation "com.facebook.stetho:stetho-okhttp3:1.6.0"
 
-    implementation 'com.github.z-chu.RxCache:rxcache:2.3.5'
-    implementation 'com.github.z-chu.RxCache:rxcache-kotlin:2.3.5'
+    implementation "com.github.z-chu.RxCache:rxcache:2.3.5"
+    implementation "com.github.z-chu.RxCache:rxcache-kotlin:2.3.5"
 
 
     if (source_code.toBoolean()) {
-        globalImplementation project(path: ':thirdlib')
+        globalImplementation project(path: ":thirdlib")
     } else {
-        globalImplementation 'com.engineer.third:thirdlib:1.0.0'
+        globalImplementation "com.engineer.third:thirdlib:1.0.0"
     }
-    implementation project(path: ':common')
-    implementation project(path: ':compose')
-    implementation 'androidx.navigation:navigation-fragment-ktx:2.7.1'
-    implementation 'androidx.navigation:navigation-ui-ktx:2.7.1'
-    implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.6.1'
+    implementation project(path: ":common")
+    implementation project(path: ":compose")
+    implementation "androidx.navigation:navigation-fragment-ktx:2.7.2"
+    implementation "androidx.navigation:navigation-ui-ktx:2.7.2"
+    implementation "androidx.lifecycle:lifecycle-runtime-ktx:2.6.2"
 
-//    debugImplementation 'com.squareup.leakcanary:leakcanary-android:2.10'
+//    debugImplementation "com.squareup.leakcanary:leakcanary-android:2.10"
 
-    implementation 'com.google.android.flexbox:flexbox:3.0.0'
+    implementation "com.google.android.flexbox:flexbox:3.0.0"
 
-    implementation 'com.alibaba:fastjson:1.1.73.android'
-    implementation 'org.jetbrains.kotlin:kotlin-script-runtime:1.8.22'
+    implementation "com.alibaba:fastjson:1.1.73.android"
+    implementation "org.jetbrains.kotlin:kotlin-script-runtime:1.8.22"
 
 
     // https://github.com/koral--/android-gif-drawable/tree/master
-    implementation 'pl.droidsonroids.gif:android-gif-drawable:1.2.26'
+    implementation "pl.droidsonroids.gif:android-gif-drawable:1.2.26"
     implementation "com.squareup.okhttp3:okhttp:4.11.0"
     implementation "com.squareup.okhttp3:logging-interceptor:4.11.0"
 
-    implementation 'com.github.microshow:RxFFmpeg:4.9.0-lite'
+    implementation "com.github.microshow:RxFFmpeg:4.9.0-lite"
 }
 
 apply from: file("../custom-gradle/test-dep.gradle") // unit test
